assets/schema_input.json

```diff
@@ -47,6 +54,6 @@
                 "unique": true
             }
         },
-        "required": ["sample"]
+        "required": ["sample_name"]
```

Comment:
The key `sample` should still be required as in IRIDA Next it contains the IRIDA Next identifiers. The key `sample_name` should be optional.

Reply:
ahh, I see sorry I misunderstood what was being requested

Reply:
fixed in: https://github.com/phac-nml/mikrokondo/pull/140/commits/6cb6d8c9cfc8e3efa5f8b778b62ed7cef39006b5

Additional commits were related to updating test file column names.

Reply:
`sample` is required, but you don't need to swap the item properties as in https://github.com/phac-nml/mikrokondo/commit/6cb6d8c9cfc8e3efa5f8b778b62ed7cef39006b5 it was correct before. 

**Quick Summary**
`sample` is the IRIDA-ID column of the samplesheet so it has to be unique, and is required. This is why we named the `meta` `meta.irida_id` (or in your case `meta.external_id`). 

`sample_name` is an optional column to simply rename file-outputs, or use in results so that the user can interperate them better. This is why @kylacochrane introduced [this](https://github.com/phac-nml/speciesabundance/blob/f0a639ddd09ffbbae9a8560b019f6681d64221ee/workflows/speciesabundance.nf#L72-L104) at the start of the workflow (I did it for all other pipelines too) which is basically:
```
if (!meta.id) {
    meta.id = meta.irida_id
```
This means it if it is run locally it'll default to the old "just use sample column" and not break anything.

Reply:
Thanks for the comment Steven. This is correct. Could you swap back `sample` and `sample_name` Matthew? But make it so that `sample` is the one that is required?

Reply:
Can you clarify what you mean by swap back `sample` and `sample_name`? Do you mean just within the `schema_input.json` or for the whole pipeline?

Reply:
Yes exactly! Just for the `schema_input.json`

Reply:
all is made well in this commit: https://github.com/phac-nml/mikrokondo/pull/140/commits/71260a99ab1334d0ffcafb008e673dce804c973a